internal/statetransition/state_transition.go

```diff
@@ -1641,9 +1632,8 @@ func anchorBlockInRecentBlocks(context block.RefinementContext, currentState *st
 			return false, fmt.Errorf("bad state root")
 		}
 
-		// Block found, check MMR
-		mountainRange := mountain_ranges.New()
-		beefyRoot := mountainRange.SuperPeak(y.AccumulationResultMMR, crypto.KeccakData)
+		// Block found, check beefy root
+		beefyRoot := y.BeefyRoot
```

Comment:
You don't need to do the super peak function here anymore, because that's now already part of recent history as the beefy root.